rust/connlib/shared/src/lib.rs

```diff
@@ -51,7 +51,10 @@ pub fn keypair() -> (StaticSecret, PublicKey) {
     (private_key, public_key)
 }
 
-pub fn get_user_agent(os_version_override: Option<String>) -> String {
+pub fn get_user_agent(
+    os_version_override: Option<String>,
+    firezone_package_version: &str,
```

Comment:
```suggestion
    app_version: &str,
```

Bit shorter. I am not attached to "app" but would like it to not say "firezone".

Reply:
Makes sense, "app" is generic over "Client" and "Gateway". And it's not a Firezone protocol version so that could be confusing.

Reply:
You could expose a macro for building the user-agent, then you could call `env!` inside it because macro get evaluated in the calling crate, not the defining one.

Reply:
Yeah. Think I should merge this for now so that `cargo run` works again?
It's still showing it stacked on the branch that's already merged into main... will fix that in a sec